app/controllers/discovered_hosts_controller.rb

```diff
@@ -24,11 +25,46 @@ def index
   end
 
   def show
-    # filter graph time range
     @range = nil
-
-    # summary report text
     @report_summary = nil
+    regex_hightlights = Setting[:discovery_CATEGORY_highlights].empty? ? /^(productname|memorysize|interface|manufacturer|architecture|macaddress$|processorcount|physicalprocessorcount|discovery_boot|ipaddress$)/ : Regexp.new(eval(Setting[:discovery_CATEGORY_highlights]))
+    regex_storage = Setting[:discovery_CATEGORY_storage].empty? ? /^blockdevice/ : Regexp.new(eval(Setting[:discovery_CATEGORY_storage]))
+    regex_hardware = Setting[:discovery_CATEGORY_hardware].empty? ? /^(hardw|manufacturer|memo|process)/ : Regexp.new(eval(Setting[:discovery_CATEGORY_hardware]))
+    regex_network = Setting[:discovery_CATEGORY_network].empty? ? /^(ipaddress|interfaces|dhcp|fqdn|hostname|link|mtu|net)|(#{@host.facts_hash["interfaces"].gsub(",","|")})/ : Regexp.new(eval(Setting[:discovery_CATEGORY_network]))
```

Comment:
Hmmm I'd rather keep all the regexps clean and static. I think if we copy few facts into highlights AND network, it's no issue. Just rewrite it to be simple regexp without this magic.